packages/worker/src/npmRegistry/api.ts

```diff
@@ -60,3 +90,51 @@ export async function fetchPackageStatsAndMetadata(name: string) {
     ...metadata,
   };
 }
+
+function isValidUrl(maybeUrl: string) {
+  let url;
+
+  try {
+    url = new URL(maybeUrl);
+  } catch (_) {
+    return false;
+  }
+
+  return url.protocol === 'http:' || url.protocol === 'https:';
+}
+
+export function getRepositoryUrl(repository?: string) {
+  let url;
+
+  if (!repository) {
+    return;
+  }
+
+  // (git+)https://github.com/facebook/react(.git)
+  if ((url = repository.match(/^(?:git\+)?(https?:.*?)(?:\.git)?$/))) {
```

Comment:
nit: When working with capture groups, destructuring can bring clarity to what's being extracted, like:
```
const urlMatch = repository.match(/^(?:git\+)?(https?:.*?)(?:\.git)?$/);
if (urlMatch) {
  const [_, reporsitoryUrl] = urlMatch;
  ...
}
```

Reply:
good point, fixed